ext/standard/var_unserializer.re

```diff
@@ -644,11 +644,13 @@ declared_property:
 					if (UNEXPECTED(obj->ce->ce_flags & ZEND_ACC_NO_DYNAMIC_PROPERTIES)) {
 						zend_throw_error(NULL, "Cannot create dynamic property %s::$%s",
 							ZSTR_VAL(obj->ce->name), zend_get_unmangled_property_name(Z_STR_P(&key)));
+						zval_ptr_dtor_str(&key);
 						goto failure;
 					} else if (!(obj->ce->ce_flags & ZEND_ACC_ALLOW_DYNAMIC_PROPERTIES)) {
 						zend_error(E_DEPRECATED, "Creation of dynamic property %s::$%s is deprecated",
 							ZSTR_VAL(obj->ce->name), zend_get_unmangled_property_name(Z_STR_P(&key)));
 						if (EG(exception)) {
+							zval_ptr_dtor_str(&key);
```

Comment:
There are a few other `goto failures` nearby where the `key` variable is not destroyed properly beforehand. As far as I an understand, these should also be fixed 🤔

Reply:
When `ret < 0` on line 636, is_property_visibility_changed would already call zval_ptr_dtor_key internally.

```
		if (UNEXPECTED(zend_unmangle_property_name_ex(Z_STR_P(key), &unmangled_class, &unmangled_prop, &unmangled_prop_len) == FAILURE)) {
			zval_ptr_dtor_str(key);
			return -1;
		}
```

The other `goto failure`  seem fine, e.g. the string key is released before unserializing the data value

Reply:
Oh, thanks for the investigation :) I didn't take a depper look into `is_property_visibility_changed()`.